exec/java-exec/src/main/java/org/apache/drill/exec/store/StoragePluginRegistry.java

```diff
@@ -34,12 +34,22 @@ public interface StoragePluginRegistry extends Iterable<Map.Entry<String, Storag
 
   @SuppressWarnings("serial")
   public static class PluginException extends Exception {
+
+    public final StoragePlugin plugin;
```

Comment:
Is there any particular reason for having a plugin in the exception class?

Reply:
@vvysotskyi I added a reference to the plugin related to the exception so that when an exception is thrown from PlannerPhase#getStorageRules the calling code has a way to find _which_ plugin from the collection that it passed in was the one that failed. It needs to know that if it is to go on to disable that plugin. But there must be other ways of sharing that data with the caller if you think using the exception object is not a good choice...

Reply:
Looks like it is not used now